app/controllers/application_controller/ci_processing.rb

```diff
@@ -531,113 +452,131 @@ def check_compliance_vms
   end
   alias_method :instance_shelve_offload, :shelveoffloadvms
   alias_method :vm_shelve_offload, :shelveoffloadvms
 
   # Reset all selected or single displayed vm(s)
   def resetvms
     assert_privileges(params[:pressed])
-    vm_button_operation('reset', _('Reset'))
+    action = Proc.new { |ids, task, name| process_objects(ids, task, name) }
+    generic_button_operation 'reset', _('Reset'), action
   end
   alias_method :instance_reset, :resetvms
   alias_method :vm_reset, :resetvms
 
   # Shutdown guests on all selected or single displayed vm(s)
   def guestshutdown
     assert_privileges(params[:pressed])
-    vm_button_operation('shutdown_guest', _('Shutdown Guest'))
+    action = Proc.new { |ids, task, name| process_objects(ids, task, name) }
+    generic_button_operation 'shutdown_guest', _('Shutdown Guest'), action
   end
   alias_method :vm_guest_shutdown, :guestshutdown
 
   # Standby guests on all selected or single displayed vm(s)
   def gueststandby
     assert_privileges(params[:pressed])
-    vm_button_operation('standby_guest', _('Standby Guest'))
+    action = Proc.new { |ids, task, name| process_objects(ids, task, name) }
+    generic_button_operation 'standby_guest', _('Standby Guest'), action
   end
 
   # Restart guests on all selected or single displayed vm(s)
   def guestreboot
     assert_privileges(params[:pressed])
-    vm_button_operation('reboot_guest', _('Restart Guest'))
+    action = Proc.new { |ids, task, name| process_objects(ids, task, name) }
+    generic_button_operation 'reboot_guest', _('Restart Guest'), action
   end
   alias_method :instance_guest_restart, :guestreboot
   alias_method :vm_guest_restart, :guestreboot
 
   # Delete all snapshots for vm(s)
   def deleteallsnapsvms
     assert_privileges(params[:pressed])
-    vm_button_operation('remove_all_snapshots', _('Delete All Snapshots'), 'vm_common/config')
+    action = Proc.new { |ids, task, name| process_objects(ids, task, name) }
+    generic_button_operation 'remove_all_snapshots', _('Delete All Snapshots'), action,
+                             :refresh_partial => 'vm_common/config' if !@explorer
   end
   alias_method :vm_snapshot_delete_all, :deleteallsnapsvms
 
   # Delete selected snapshot for vm
   def deletesnapsvms
     assert_privileges(params[:pressed])
-    vm_button_operation('remove_snapshot', _('Delete Snapshot'), 'vm_common/config')
+    action = Proc.new { |ids, task, name| process_objects(ids, task, name) }
+    generic_button_operation 'remove_snapshot', _('Delete Snapshot'), action,
+                             :refresh_partial => 'vm_common/config' if !@explorer
   end
   alias_method :vm_snapshot_delete, :deletesnapsvms
 
   # Delete selected snapshot for vm
   def revertsnapsvms
     assert_privileges(params[:pressed])
-    vm_button_operation('revert_to_snapshot', _('Revert to a Snapshot'), 'vm_common/config')
+    action = Proc.new { |ids, task, name| process_objects(ids, task, name) }
```

Comment:
Can you use `method` rather thatn `Proc`? So that we don't create a closure each time (which is expensive in ruby). But do not spend too much time on this one ;-)

Reply:
@martinpovolny corrected, thanks!